### ISSUE TITLE
Unable to rename a variable in netCDF-4 file

### ISSUE THREAD

**Issue by gsjaardema** [Unidata/netcdf-c#2070]
The attached program attempts to rename a two-dimensional variable in a netCDF-4 format file.

It creates the file with two variables.  It then closes the file and reopens it with `NC_WRITE` and renames one of the variables with a name that is the same length as the original name.  It then closes the file and reopens it with `NC_NOWRITE` and queries the two variables on the file and then closes the file.

All of the operations succeed with no errors returned. However, if I ncdump the file at the end of the run, it is corrupt and throws an HDF error.   If I examine the file after the initial creation, it looks correct, so something seems to corrupt the file during the variable rename even though it looks like all api calls succeed.

This is related to #2047 since the only test failing for the PR is a variable rename for a 1D variable.  I wanted to see if renaming a 2D variable worked without that PR applied and found that it was corrupting the file.

[test-rename.c.txt](https://github.com/Unidata/netcdf-c/files/6969842/test-rename.c.txt)

**Comment by edwardhartnett** [comment]
Renames are a real pain.

I have an idea of a total rewrite of the rename code, solving all problems. But that cannot happen this week unfortunately...

**Comment by gsjaardema** [comment]
No, understandably.  Just using this issue to show that the test failure in #2047 is something that is not something wrong with that PR, but just illustrates a problem already existing in the code.
